misc/multihash/src/lib.rs

```diff
@@ -180,15 +204,25 @@ impl<'a> MultihashRef<'a> {
         let (code, bytes) = decode::u16(&input).map_err(|_| DecodeError::BadInputLength)?;
 
         let alg = Hash::from_code(code).ok_or(DecodeError::UnknownCode)?;
+
+        // handle the identity case
+        if alg == Hash::Identity {
+            let (hash_len, bytes) = decode::u32(&bytes).map_err(|_| DecodeError::BadInputLength)?;
+            if bytes.len() as u32 != hash_len {
```

Comment:
Again using `as_u64` from above:

```suggestion
            if as_u64(bytes.len()) != u64::from(hash_len) {
```